presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestMergeAdjacentWindows.java

```diff
@@ -114,8 +115,49 @@ public void testDoesNotFireForNonWindowIntermediateNode()
                                                 newWindowNodeSpecification(p, "a"),
                                                 ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", "a")),
                                                 p.values(p.symbol("a", BIGINT), p.symbol("b", BIGINT))
+                                        )
                                 )
-                        )))
+                        ))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDependentAdjacentWindowsIdenticalSpecifications()
+            throws Exception
+    {
+        Optional<Window> windowA = Optional.of(new Window(ImmutableList.of(new SymbolReference("a")), ImmutableList.of(), Optional.empty()));
+
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p ->
+                        p.window(
+                                newWindowNodeSpecification(p, "a"),
+                                ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", windowA, "avg_2")),
+                                p.window(
+                                        newWindowNodeSpecification(p, "a"),
+                                        ImmutableMap.of(p.symbol("avg_2", BIGINT), newWindowNodeFunction("avg", windowA, "a")),
+                                        p.values(p.symbol("a", BIGINT), p.symbol("b", BIGINT))
+                                )
+                        ))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDependentAdjacentWindowsDistinctSpecifications()
```

Comment:
There are three ways how windows can depends on.
Windows can depend when one is using child symbols in either function call, order by or partition by. 
I see that you added tests which are only using symbol in function calls, what about testing when symbols are used in partition by or order by?